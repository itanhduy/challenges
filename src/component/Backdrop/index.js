import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'

const BackdropComponent = styled.div`
  ${props => {
    const { theme, styleName } = props
    const { createStyle, color, convert } = theme
    return {
      position: 'fixed',
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      backgroundColor: convert.colorOpacity(color.text, 0.9),
      ...createStyle(styleName, 'backdrop'),
    }
  }};
`

class Backdrop extends PureComponent {
  render() {
    const { children } = this.props
    return (
      <BackdropComponent {...this.props} styleName="fadeIn flexible v-center h-center">
        {children}
      </BackdropComponent>
    )
  }
}

Backdrop.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
}

export default withTheme(Backdrop)
