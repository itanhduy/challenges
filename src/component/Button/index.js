import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'
import Text from '../Text'

const ButtonComponent = styled.div`
  ${props => {
    const { theme, styleName } = props
    const { color, createStyle } = theme
    return {
      border: `2px solid ${color.primary}`,
      padding: '5px',
      cursor: 'pointer',
      ...createStyle(styleName, 'button'),
    }
  }};
`

class Button extends PureComponent {
  render() {
    const { children, textProps, styleName } = this.props
    return (
      <ButtonComponent {...this.props} styleName={`borderRadius h-center flexible ${styleName}`}>
        <Text {...textProps}>{children}</Text>
      </ButtonComponent>
    )
  }
}

Button.propsType = {
  styleName: PropTypes.string,
  children: PropTypes.any,
  textProps: PropTypes.object,
}

Button.defaultProps = {
  styleName: String(),
}

export default withTheme(Button)
