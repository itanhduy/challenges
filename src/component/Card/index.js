import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'
import Image from '../Image'
import Text from '../Text'

const CardWrapper = styled.div`
  ${props => {
    const { columns, theme, styleName } = props
    const { createStyle } = theme
    return {
      width: `${100 / columns}%`,
      ...createStyle(styleName, 'card'),
    }
  }};
`

const CardComponent = styled.div`
  ${props => {
    const { theme, cardPadding, styleName } = props
    const { createStyle } = theme
    return {
      width: `calc(100% - ${cardPadding}px)`,
      overflow: 'hidden',
      ...theme.general.boxShadow,
      ...createStyle(styleName, 'card'),
    }
  }};
`

const CardContent = styled.div`
  ${props => {
    const { theme, styleName } = props
    const { createStyle } = theme
    return {
      padding: '25px 15px',
      ...createStyle(styleName, 'card'),
    }
  }};
`

class Card extends PureComponent {
  render() {
    const { data } = this.props
    return (
      <CardWrapper {...this.props} styleName="flexible v-center">
        <CardComponent {...this.props} styleName="borderRadius">
          <Image url={data.image} height={200} />
          <CardContent>
            <Text styleName="medium">{data.name}</Text>
          </CardContent>
        </CardComponent>
      </CardWrapper>
    )
  }
}

Card.propTypes = {
  columns: PropTypes.number.isRequired,
  cardPadding: PropTypes.number,
  data: PropTypes.object,
}

Card.defaultProps = {
  columns: 2,
  cardPadding: 15,
}

export default withTheme(Card)
