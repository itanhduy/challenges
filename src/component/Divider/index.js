import React, { PureComponent } from 'react'
import styled, { withTheme } from 'styled-components'

const DividerComponent = styled.div`
  ${props => {
    const { theme, styleName } = props
    const { color, createStyle } = theme
    return {
      borderBottom: `1px solid ${color.border}`,
      opacity: 0.5,
      ...createStyle(styleName, 'divider'),
    }
  }};
`

class Divider extends PureComponent {
  render() {
    return <DividerComponent styleName="width-100" />
  }
}

export default withTheme(Divider)
