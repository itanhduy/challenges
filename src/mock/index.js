import DonateMock from './donate.mock'

export { DonateMock }
