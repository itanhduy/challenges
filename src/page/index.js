import Donate from './Donate'
import Home from './Home'
import ThankYou from './ThankYou'

export { Donate, Home, ThankYou }
