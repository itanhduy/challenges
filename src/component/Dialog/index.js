import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'
import Backdrop from '../Backdrop'
import Screen from '../Screen'
import Row from '../Row'
import Text from '../Text'
import Button from '../Button'

const DialogComponent = styled.div`
  ${props => {
    const { theme, styleName } = props
    const { color, createStyle } = theme
    return {
      position: 'absolute',
      backgroundColor: color.white,
      ...createStyle(styleName, 'dialog'),
    }
  }};
`

class Dialog extends PureComponent {
  componentDidMount() {
    /**
     * Remove scroll bar from body element
     */
    document.body.style.overflow = 'hidden'
  }

  componentWillUnmount() {
    /**
     * Rollback default styling for body element
     */
    document.body.style.removeProperty('overflow')
  }

  /**
   * Render button at the bottom
   */
  renderButtons = () => {
    const { onClickOK, onClickClose, showOKButton, showCloseButton } = this.props
    let rowWrapperStyleName = 'xl-gutter-top flexible v-center h-center'

    if (showOKButton && showCloseButton) {
      rowWrapperStyleName += ' space-between'
    }

    return (
      <Row styleName={rowWrapperStyleName}>
        {showOKButton && (
          <Button styleName="width-45" textProps={{ styleName: 'textPrimary medium' }} onClick={onClickOK}>
            OK
          </Button>
        )}
        {showCloseButton && (
          <Button styleName="width-45" textProps={{ styleName: 'textPrimary medium' }} onClick={onClickClose}>
            Close
          </Button>
        )}
      </Row>
    )
  }

  /**
   * Render dialog with show, title, description
   * @return {PureComponent} The dialog component
   */
  renderDialog = () => {
    const { show, title, description } = this.props
    if (show) {
      return (
        <Backdrop>
          <DialogComponent {...this.props} styleName="bounceIn xl-gutter borderRadius">
            <Screen>
              <Row styleName="h-center v-center">
                <Text styleName="bold">{title}</Text>
              </Row>
              <Row styleName="md-gutter-top h-center v-center">
                <Text styleName="medium">{description}</Text>
              </Row>
              {this.renderButtons()}
            </Screen>
          </DialogComponent>
        </Backdrop>
      )
    }
    return null
  }

  render() {
    return this.renderDialog()
  }
}

Dialog.propTypes = {
  show: PropTypes.bool,
  title: PropTypes.string,
  description: PropTypes.string,
  onClickOK: PropTypes.func,
  onClickClose: PropTypes.func,
  showOKButton: PropTypes.bool,
  showCloseButton: PropTypes.bool,
}

Dialog.defaultProps = {
  onClickOK: () => {},
  onClickClose: () => {},
  showOKButton: true,
  showCloseButton: true,
  show: false,
}

export default withTheme(Dialog)
