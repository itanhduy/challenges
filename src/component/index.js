import Button from './Button'
import Card from './Card'
import Image from './Image'
import ListCharities from './ListCharities'
import Row from './Row'
import Screen from './Screen'
import Text from './Text'
import Divider from './Divider'
import PaymentOptions from './PaymentOptions'
import Dialog from './Dialog'
import Nothing from './Nothing'
import GoBackHome from './GoBackHome'
import ProviderProvide from './ProviderProvide'
import Backdrop from './Backdrop'

export {
  Card,
  ListCharities,
  Screen,
  Text,
  Row,
  Button,
  Image,
  Divider,
  PaymentOptions,
  Dialog,
  Nothing,
  GoBackHome,
  ProviderProvide,
  Backdrop,
}
