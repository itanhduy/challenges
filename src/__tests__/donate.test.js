import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { DonateMock } from '../mock'
import { ProviderProvide } from '../component'
import { Donate } from '../page'
import renderer from 'react-test-renderer'

test('Should render Donate component', () => {
  const donateComponent = renderer.create(
    <ProviderProvide>
      <BrowserRouter>
        <Donate {...DonateMock} />
      </BrowserRouter>
    </ProviderProvide>,
  )
  const tree = donateComponent.toJSON()
  expect(tree).toMatchSnapshot()
})
