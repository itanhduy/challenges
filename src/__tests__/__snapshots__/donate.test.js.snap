// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Donate component 1`] = `
<div
  className="Screen__ScreenComponent-sc-182dw2i-0 ksPqLC"
>
  <div
    className="Row__RowComponent-sc-6b1omn-0 ilGeGH"
  >
    <div
      className="Text__TextComponent-sc-1bxt3zu-0 iimCvt"
    >
      Donation for 
    </div>
  </div>
  <div
    className="Row__RowComponent-sc-6b1omn-0 gxdfZP"
  >
    <div
      className="Card__CardWrapper-sc-9nsasl-0 drlepc"
      data={Object {}}
    >
      <div
        className="Card__CardComponent-sc-9nsasl-1 clGIEO"
        data={Object {}}
      >
        <div
          className="Image__ImageComponent-sc-1qzvlhi-0 cIPFke"
          height={200}
        />
        <div
          className="Card__CardContent-sc-9nsasl-2 jbOzcf"
          data={Object {}}
        >
          <div
            className="Row__RowComponent-sc-6b1omn-0 dPagVA"
          >
            <div
              className="Text__TextComponent-sc-1bxt3zu-0 bcXjgf"
            />
          </div>
        </div>
        <div
          className="Card__CardContent-sc-9nsasl-2 giNtnB"
        >
          <div
            className="Row__RowComponent-sc-6b1omn-0 dPagVA"
          >
            <div
              className="Divider__DividerComponent-sc-1m9b2q8-0 bSmuG"
            />
            <div
              className="Row__RowComponent-sc-6b1omn-0 ctwCvs"
            >
              <div
                className="Text__TextComponent-sc-1bxt3zu-0 bcXjgf"
              >
                Payment options
              </div>
              <div
                className="Row__RowComponent-sc-6b1omn-0 ctwCvs"
              />
            </div>
            <div
              className="Row__RowComponent-sc-6b1omn-0 jieBAG"
            >
              <div
                className="Divider__DividerComponent-sc-1m9b2q8-0 bSmuG"
              />
            </div>
            <div
              className="Row__RowComponent-sc-6b1omn-0 jieBAG"
            >
              <div
                className="Text__TextComponent-sc-1bxt3zu-0 bcXjgf"
              >
                Please choose at least one option
              </div>
            </div>
            <div
              className="Row__RowComponent-sc-6b1omn-0 fLhEPp"
            >
              <div
                className="Button__ButtonComponent-sc-1b5ft91-0 gDPDbt"
                onClick={[Function]}
              >
                <div
                  className="Text__TextComponent-sc-1bxt3zu-0 khpdy"
                >
                  Donate
                </div>
              </div>
              <div
                className="Button__ButtonComponent-sc-1b5ft91-0 gDPDbt"
                onClick={[Function]}
              >
                <div
                  className="Text__TextComponent-sc-1bxt3zu-0 khpdy"
                >
                  Go Back Home
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
